### PR TITLE
Add cmd history-exporter

### PR DESCRIPTION
**Pull request by fedordikarev**
- Rearrange modules and make libraries reuasable
- We want to have historical data analysis as well, for that create `history-exporter` that will import historical data
- Disable docker push in ci for now, reenable it later

### CODE REVIEW

**Comment by fedordikarev** [comment]
omg, what I did here:
1. I list workflows runs for a given period of time
2. from that list I get id and attempt number
3. if information for that id and attempt is already in DB, I skip it
4. if not, I query gh api one more time for the info and save it into DB.

you also see it?
I already have all the information needed for the step 4 in results of step 1 (not sure yet, have to check it).
And if it's true, I could not only speed up that process, but also stop hitting rate limit for GH API, so "month long export" could be done in matter of tens of minutes, not tens of hours as it is right now.

File: .github/workflows/ci.yml
Comment (bayandin) [inline .github/workflows/ci.yml:1]: We won't need changes this file once https://github.com/neondatabase/gh-workflow-stats-action/pull/9 is merged

File: pkg/export/export.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+func ExportAndSaveJobs(ctx context.Context, conf config.ConfigType, runAttempt int64) error {
+	jobsInfo, rate, err := gh.GetWorkflowAttemptJobs(ctx, conf, runAttempt)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if rate.Remaining < 20 {
+		fmt.Printf("Close to rate limit, remaining: %d", rate.Remaining)
+		fmt.Printf("Sleep till %v (%v seconds)\n", rate.Reset, time.Until(rate.Reset.Time))
+		time.Sleep(time.Until(rate.Reset.Time) + 10*time.Second)
```
Comment (bayandin) [inline pkg/export/export.go:22]: It seems the comment slightly doesn't match the code itself (10s difference)

File: cmd/history-exporter/main.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+	flag.StringVar(&startDateStr, "start-date", "", "start date to quert and export")
+	flag.StringVar(&endDateStr, "end-date", "", "end date to quert and export")
+	flag.Parse()
+
+	if startDateStr == "" {
+		startDate = time.Now().Truncate(24 * time.Hour)
+	} else {
+		var err error
+		startDate, err = time.Parse("2006-01-02", startDateStr)
```
Comment (bayandin) [inline cmd/history-exporter/main.go:31]: Just curious, why `2006-01-02`?

**Review by bayandin — APPROVED** [review]
I gave a brief look, so if it works — it works :) 

Thanks!

File: cmd/history-exporter/main.go (hunk 2 above)
Comment (fedordikarev) [reply cmd/history-exporter/main.go:31]: it's "Format" string to parse the date in Golang, and I also often confused about it, especially when working with other languages.
so here "2006" encodes "parse year with 4 digits" (analog of "%Y" in other languages), "01" means "2 digits month" ("%m" analog) and "02" means "2 digits day" ("%d" analog)
https://github.com/golang/go/blob/master/src/time/format.go#L152 with full format definitions

File: .github/workflows/ci.yml
Comment (fedordikarev) [reply .github/workflows/ci.yml:1]: updated, thanks!

File: pkg/export/export.go (hunk 1 above)
Comment (fedordikarev) [reply pkg/export/export.go]: updated, thanks!

**Pull request merged**